File: UniShare.API/Models/DTOs/Users/UpdateProfileRequest.cs
namespace UniShare.API.Models.DTOs.Users;

public class UpdateProfileRequest
{
    public string? FullName { get; set; }
    public string? PhoneNumber { get; set; }
    public string? AvatarUrl { get; set; }
    public Guid? SchoolId { get; set; }
    public Guid? AreaId { get; set; }
}

File: UniShare.APP/lib/features/users/models/update_profile_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'update_profile_request.g.dart';

@JsonSerializable()
class UpdateProfileRequest {
  final String fullName;
  final String? phoneNumber;
  final String? avatarUrl;
  final String? schoolId;
  final String? areaId;

  const UpdateProfileRequest({
    required this.fullName,
    this.phoneNumber,
    this.avatarUrl,
    this.schoolId,
    this.areaId,
  });

  factory UpdateProfileRequest.fromJson(Map<String, dynamic> json) =>
      _$UpdateProfileRequestFromJson(json);

  Map<String, dynamic> toJson() => _$UpdateProfileRequestToJson(this);
}
